controller.go

```diff
@@ -234,3 +267,75 @@ func ensureNodeExists(instances cloudprovider.Instances, node *v1.Node) (bool, e
 
 	return true, nil
 }
+
+// If clusterAPI and clusterID are set, then we are running
+// inside Giantnetes, so try to get certificates and create
+// kubernetes client for guest cluster. If not proceed with
+// kubeconfig file.
+func newGuestClient(clusterID, clusterAPI, master, kubeconfig string) (clientset.Interface, error) {
+	if clusterAPI != "" && clusterID != "" {
+		return newGuestClientFromSecret(clusterAPI, clusterID)
+	}
+
+	if kubeconfig == "" {
+		err := fmt.Errorf("kubeconfig must be set, if %v and %v are not used", clusterAPIEnv, clusterIDEnv)
+		return nil, err
+	}
+
+	// Create kubeclient config from kubeconfig file.
+	config, err := clientcmd.BuildConfigFromFlags(master, kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create the client.
+	kubeClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return kubeClient, nil
+}
+
+func newGuestClientFromSecret(clusterID, clusterAPI string) (clientset.Interface, error) {
+	// Get host cluster client to retrieve certificates.
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	// creates the clientset
+	hostClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Get API secret with certificates.
+	secretName := fmt.Sprintf("%s-api", clusterID)
+
+	secret, err := hostClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// XXX(r7vme): Not using operatorkit k8s client here, because it does not support
+	// TLS certificates as raw data, only supports files.
```

Comment:
I think raw data should be the default supported way. We should change that. https://github.com/giantswarm/operatorkit/issues/97